From 9c8db36f9ff056dbc33a33ef836442e13ac8c9b3 Mon Sep 17 00:00:00 2001
From: Piotr Nakraszewicz <Piotr.Nakraszewicz@tieto.com>
Date: Sat, 11 Feb 2012 10:54:40 +0200
Subject: [PATCH 30/75] P2P: Do not expire GO peer entry during group rekeying

If wpas_go_connected() is called during group rekeying the P2P GO peer
will expire. To prevent that check if group rekeying is not in progress.
(cherry picked from commit d9a0f666960ee096fbaf21e72818772dbfa043b7)
---
 wpa_supplicant/p2p_supplicant.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/wpa_supplicant/p2p_supplicant.c
+++ b/wpa_supplicant/p2p_supplicant.c
@@ -2219,7 +2219,8 @@ static int wpas_go_connected(void *ctx,
 			continue;
 		if (ssid->mode != WPAS_MODE_INFRA)
 			continue;
-		if (wpa_s->wpa_state != WPA_COMPLETED)
+		if (wpa_s->wpa_state != WPA_COMPLETED &&
+		    wpa_s->wpa_state != WPA_GROUP_HANDSHAKE)
 			continue;
 		if (os_memcmp(wpa_s->go_dev_addr, dev_addr, ETH_ALEN) == 0)
 			return 1;
